store/postgres/src/relational/index.rs

```diff
@@ -517,13 +563,137 @@ impl CreateIndex {
             ) => {
                 unique == o_unique
                     && method == o_method
-                    && columns == o_columns
+                    && Expr::is_same_kind_columns(columns, o_columns)
                     && cond == o_cond
                     && with == o_with
             }
         }
     }
 
+    pub fn is_id_immutable_table(&self, immutable: bool) -> bool {
+        // on imutable tables the id constraint is specified at table creation
+        if immutable {
+            match self {
+                CreateIndex::Unknown { .. } => (),
+                CreateIndex::Parsed { columns, .. } => {
+                    if columns.len() == 1 {
+                        if columns[0].is_id() {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        false
+    }
+
+    pub fn to_postpone(&self) -> bool {
+        fn has_prefix(s: &str, prefix: &str) -> bool {
+            s.starts_with(prefix)
+                || s.ends_with("\"") && s.starts_with(format!("\"{}", prefix).as_str())
+        }
+        match self {
+            CreateIndex::Unknown { .. } => false,
+            CreateIndex::Parsed {
+                name,
+                columns,
+                method,
+                ..
+            } => {
+                if *method != Method::BTree && *method != Method::Brin {
```

Comment:
I don't think we want to postpone BRIN indexes. They are small and cheap to build; we probably don't do that much, but if we turned the account-like optimization on during indexing, the BRIN index could help speed up lookups of mutable entities.